extra_types.h

```diff
@@ -5,5 +5,5 @@ typedef unsigned long long xulong;
 typedef long long xlong;
 #else
 typedef unsigned long xulong;
```

Comment:
This doesn't seem right, let me check. Think this will break a whole lot of stuff.